sessionctx/variable/varsutil.go

```diff
@@ -156,6 +162,198 @@ func ValidateGetSystemVar(name string, isGlobal bool) error {
 	return nil
 }
 
+// ValidateSetSystemVar checks if system variable satisfies specific restriction.
+func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string, error) {
+	if strings.EqualFold(value, "DEFAULT") {
+		if val := GetSysVar(name); val != nil {
+			return val.Value, nil
+		}
+		// should never happen
+		panic(fmt.Sprintf("Error happened when ValidateSetSystemVar. Invalid system variable: %s", name))
```

Comment:
Will it be more friendly if we return UnknownSystemVar.GenByArgs(name) here

Reply:
ok